### PR TITLE
Handle userdata on Windows

### PR DESCRIPTION
**Pull request by flingure**

### CODE REVIEW

File: e3/aws/cfn/ec2/__init__.py [hunk 1]
```diff
@@ -173,6 +177,40 @@ def properties(self):
...
+    def properties(self):
+        """Serialize the object as a simple dict.
+
+        Can be used to transform to CloudFormation Yaml format.
+
+        :rtype: dict
+        """
+        props = []
+        for kind, part in self.parts:
+            prop = ['<%s>' % kind]
```
Comment (Nikokrock) [inline e3/aws/cfn/ec2/__init__.py]: prop = [ , , ] no need to do += all the time no

File: e3/aws/cfn/ec2/__init__.py [hunk 2]
```diff
@@ -173,6 +177,40 @@ def properties(self):
...
+            prop += ["%s" % part]
+            prop += ['</%s>' % kind]
+            props.append("\n".join(prop))
+        return {"Fn::Base64": {"Fn::Join": ["", props]}}
```
Comment (Nikokrock) [inline e3/aws/cfn/ec2/__init__.py]: Use Base64 and Join from e3.aws.cfn

File: e3/aws/cfn/ec2/__init__.py [hunk 3]
```diff
@@ -260,7 +298,8 @@ def set_cfn_init(self, stack,
                      cfn_init='/usr/local/bin/cfn-init',
                      region=None,
                      resource=None,
-                     metadata=None):
+                     metadata=None,
+                     kind='linux'):
```
Comment (Nikokrock) [inline e3/aws/cfn/ec2/__init__.py]: kind can be deduced from AMI information (in self.image)

Comment (Nikokrock) [reply e3/aws/cfn/ec2/__init__.py]: self.image.data.get('Platform', 'linux')   (not always defined for linux)

File: e3/aws/cfn/ec2/__init__.py [hunk 4]
```diff
@@ -308,6 +358,11 @@ def add_user_data(self, name, kind, content):
             self.user_data = UserData()
         self.user_data.add(name, kind, content)
 
+    def add_win_user_data(self, kind, content):
```
Comment (Nikokrock) [inline e3/aws/cfn/ec2/__init__.py]: use add_user_data and dispatch based on the platform information from the ami (probably we want an attribute:

self.is_windows = self.image.data.get('Platform', 'linux')

File: e3/aws/cfn/ec2/__init__.py [hunk 5]
```diff
@@ -173,6 +177,40 @@ def properties(self):
...
+        :param content: the content associated with that value
+        :type content: str
+        """
+        self.parts.append((kind, content))
+
+    @property
+    def properties(self):
+        """Serialize the object as a simple dict.
+
+        Can be used to transform to CloudFormation Yaml format.
```
Comment (quinot) [inline e3/aws/cfn/ec2/__init__.py]: Ummm.  Looks to me like we're producing XML, not YaML. Also, we're building it by hand, aren't there existing modules that we can reuse for this?

**Pull request closed without merging**